### ISSUE TITLE
Access location only while in use, in Android

### ISSUE THREAD

**Issue by fbambusi** [Lyokone/flutterlocation#428]
## 🚀 Feature Requests

Good morning everyone! Google has updated Permissions & Location Permissions policies applications must comply with to be published on [Google Play](https://play.google.com/store).

The new policy restricts the possibility to access users' location in the background: it would be great to have the possibility to access the location in the foreground only.

In particular, it is now necessary to request the ACCESS_BACKGROUND_LOCATION permission in the manifest, and to follow the guidelines provided [here](https://support.google.com/googleplay/android-developer/answer/9799150?hl=en&ref_topic=2364761).

In case the application is not compliant with the new policy, updates are rejected with error messages similar to this one
`Background location access not declared
We detected that your app contains at least one feature that requests background location access, however you have not submitted a permission declaration form for this feature. Please log in to your Play Console and submit a declaration form. `

### Contextualize the feature
There should exist two new functionalities, which could be implemented either by means of new functions or by adding parameters to the existing ones:

1. Request foreground location only
2. Access foreground location, if available, and raise an exception else.

### Describe the feature
With the proposed modification, the framework should access users' location only when the application is active, without performing any kind of activity when the app is put in background or turned off.

### Platforms affected (mark all that apply)
- [ ] :iphone: iOS
- [x] :robot: Android

**Comment by eSDhee** [comment]
I got this issue while make a new release. I think what @fbambusi asked for is really needed now!

![image](https://user-images.githubusercontent.com/960958/96884499-c3b03400-14ab-11eb-9796-e866918ee35d.png)

**Comment by keronei** [comment]
Hello,
You might want to consider this: In the release of Android 11, google announced that in order to access background location, you will be required to complete a number of extra steps, this is noted in [11 weeks of Android](https://android-developers.googleblog.com/2020/06/11-weeks-of-android-privacy-and-security.html) under **Security & Privacy**. 

In order to access background location prior to android 11(API 30), you only needed to add [ACCESS_COARSE_LOCATION](https://developer.android.com/reference/android/Manifest.permission#ACCESS_COARSE_LOCATION) permission  or [ACCESS_FINE_LOCATION](https://developer.android.com/reference/android/Manifest.permission#ACCESS_FINE_LOCATION) permission and you can handle the rest in your code.

However, going forward, one is required to declare [ACCESS_BACKGROUND_LOCATION](https://developer.android.com/training/location/receive-location-updates#request-background-location) in the manifest, in order to access the same.

In PlayStore console, you are required to target the latest Android release([Android 11, API 30 as at now](https://www.android.com/android-11/)) for every app that you publish, if your app has `targetSdkVersion` set to anything below 30, Play Console will make the assumptions that you can still access location in the background with the initial two permissions without adding `ACCESS_BACKGROUND_LOCATION` permission to the manifest and will probe for further details.

![Screenshot 2020-10-27 at 13 14 57](https://user-images.githubusercontent.com/17080971/97299569-2e73bd80-1866-11eb-800b-48aba87be55c.png)

While filling the permission form, you can specify how your app utilises the location updates in the background and you _may_ be allowed to publish, however, if you answer NO in the questionnaire, console will not allow you to proceed while insisting that you should fill the questionnaire.

![Screenshot 2020-10-27 at 13 15 15](https://user-images.githubusercontent.com/17080971/97299770-8a3e4680-1866-11eb-9eae-f35d4e948891.png)

In your scenario, you do not require access to location in the background, you can set your `targetSdkVersion` to 30, from here you will be required to declare `ACCESS_BACKGROUND_LOCATION` in order to have access to it, [without this permission in the manifest](https://developer.android.com/training/location/background), Play Console knows well that you will not use location in the background and you can go ahead and publish your app.

From my perspective, this may not have much to do with the plugin but the android privacy enforcements mechanisms.

![Screenshot 2020-10-27 at 13 18 13](https://user-images.githubusercontent.com/17080971/97301293-a7741480-1868-11eb-8726-6515d4ba863c.png)


In short, with `targetSdkVersion 30`, You can smoothly complete the process of publishing your app, be sure to also have `compileSdkVersion` set to the same value because it is required that the `compileSdkVersion` not be lower than `targetSdkVersion`.

**Comment by Pitazzo** [comment]
Hey @keronei , if targeting API level 29 is a requirement, how should we handle this issue? We are not interested in location in the background. Shall we just remove the `ACCESS_FINE_LOCATION` permission?

**Comment by keronei** [comment]
Once your app targets 29(at least for now), it’s up to you to decide on whether you need coarse or fine location.

Background location won’t be requested once you have this set unless you have explicitly requested it. If you’re still facing any error/warning and you are using Android Studio, you can open the android folder as an app on its own, then let it sync and check the app/gradle file if there’s any warning.
